File: FirebaseDB/app/src/main/java/com/example/lee_chen/firebasedb/UserAdapter.java
package com.example.lee_chen.firebasedb;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.ArrayList;

public class UserAdapter extends RecyclerView.Adapter<UserAdapter.ViewHolder> {

    private ArrayList<User> userArrayList;



    class ViewHolder extends RecyclerView.ViewHolder {
        TextView nameTextView;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            nameTextView = itemView.findViewById(R.id.name);
        }
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        View view = LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.custom_cell, viewGroup,false);
        ViewHolder viewHolder = new ViewHolder(view);
        return viewHolder;
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder viewHolder, int i) {
        User user = userArrayList.get(i);
        viewHolder.nameTextView.setText(user.getName());

    }

    @Override
    public int getItemCount() {
        if (userArrayList.size() == 0) {
            return 0;
        }
        return userArrayList.size();
    }

    public void setUserArrayList(ArrayList<User> userArrayList) {
        this.userArrayList = userArrayList;
        notifyDataSetChanged();
    }
}

File: KotlinRetrofit/app/src/main/java/com/lihan/kotlinretrofit/UserAdapter.kt
package com.lihan.kotlinretrofit

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView

class UserAdapter(private val users : ArrayList<User>) : RecyclerView.Adapter<UserViewHolder>(){
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): UserViewHolder {
        return UserViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.item_users,parent,false))
    }
    override fun getItemCount(): Int = if (users.size==0) 0 else users.size
    override fun onBindViewHolder(holder: UserViewHolder, position: Int) {
        holder.bindTo(users[position])
        holder.itemView.tag = position

    }

}